js/upgrade.js

```diff
@@ -496,27 +513,37 @@ function doAjaxRequest(action, nextParams) {
         }
       }
     },
-    error: function(jqXHR, textStatus, errorThrown) {
+    error: function (jqXHR, textStatus, errorThrown) {
       $("#pleaseWait").hide();
       if (textStatus === "timeout") {
         if (action === "download") {
           updateInfoStep(input.translation.cannotDownloadFile);
         } else {
-          updateInfoStep("[Server Error] Timeout: " + input.translation.downloadTimeout);
+          updateInfoStep(
+            "[Server Error] Timeout: " + input.translation.downloadTimeout,
+          );
+        }
+      } else {
+        try {
+          res = $.parseJSON(jqXHR.responseText);
+          addQuickInfo(res.nextQuickInfo);
+          addError(res.nextErrors);
+          updateInfoStep(res.next_desc);
+        } catch (e) {
+          updateInfoStep(
+            "[Ajax / Server Error for action " +
+              action +
+              '] textStatus: "' +
+              textStatus +
+              ' " errorThrown:"' +
+              errorThrown +
+              ' " jqXHR: " ' +
+              jqXHR.responseText +
+              '"',
```

Comment:
this part is not really readable

Reply:
This is completely rework in https://github.com/PrestaShop/autoupgrade/pull/830/files.

I put the PR in blocks to wait for the other JS PRs to merge before and restart a lint after